### PR TITLE
add pre-commit-ci code-spell check

### PR DESCRIPTION
**Pull request by XuehaoSun**
## Type of Change

Add code spell check
bug fix

## Description

[codespell](https://github.com/codespell-project/codespell)

## Expected Behavior & Potential Risk

no

## How has this PR been tested?

ci

## Dependency Change?

no

### CODE REVIEW

**Comment by chensuyue** [comment]
@ftian1 please help to check the parameter name spell issue https://results.pre-commit.ci/run/github/281528773/1693979431.WbgvE1S9QzK2bDzMxui_kg

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -107,6 +103,13 @@ repos:
         args: [--line-length=120, --skip-errors]
         exclude: docs/source-app
 
+  - repo: https://github.com/codespell-project/codespell
```
Comment (chensuyue) [inline .pre-commit-config.yaml:106]: Seems this check cover all the files including readme, let remove the Azure spell check pipeline.   
And remove the invalid words in the inc_dict.txt.

Comment (XuehaoSun) [reply .pre-commit-config.yaml:106]: > Seems this check cover all the files including readme, let's remove the Azure spell check pipeline. And remove the invalid words in the inc_dict.txt.

The current detection directory does not contain the `./examples` folder

Comment (chensuyue) [reply .pre-commit-config.yaml:106]: Can we also add similar scope like spell check `${REPO_DIR}/examples/**/*.md|!${REPO_DIR}/examples/pytorch/**/huggingface_models/**/*.md|!${REPO_DIR}/examples/README.md`?     
Use another PR to do the change, and remove the Azure spell check pipeline.

**Review by chensuyue — APPROVED** [review]

**Pull request merged**